internal/app/synchronizer/evaluator_client.go

```diff
@@ -119,10 +119,12 @@ func (ec *grcpEvaluatorClient) evaluate(ctx context.Context, pc <-chan []*pb.Mat
 	}
 
 	results := []string{}
+	matchIDs := map[string]bool{}
 
 	eg.Go(func() error {
 		for proposals := range pc {
 			for _, proposal := range proposals {
+				matchIDs[proposal.GetMatchId()] = true
```

Comment:
When checking for duplicate ids within the synchronizer, I think this is a reasonable place to do it:
- You have a set of previously sent ids.
- You're already checking for duplication.

Just check here if it's already been seen, and return an error if it has?